packages/@react-spectrum/button/test/Button.test.js

```diff
@@ -273,6 +281,42 @@ describe('Button', function () {
     expect(eventDown.defaultPrevented).toBeFalsy();
   });
 
+  // iPending state
+  it('displays a spinner after a short delay when isPending prop is true', function () {
+    let spinnerVisibilityDelay = 1000;
+    let onPressSpy = jest.fn();
+    function TestComponent() {
+      let [pending, setPending] = useState(false);
+      return (
+        <Button onPress={(pending) => { setPending(true); onPressSpy(); }} isPending={pending}>
+          Click Me
+        </Button>
+      );
+    }
+    let {getByRole} = render(<TestComponent />);
+    let button = getByRole('button');
+    expect(button).not.toHaveAttribute('aria-disabled');
+    triggerPress(button);
+    // Button is disabled immediately, but spinner visibility is delayed
+    expect(button).toHaveAttribute('aria-disabled', 'true');
+    // Multiple clicks shouldn't call onPressSpy
+    triggerPress(button);
```

Comment:
any reason not to check that the spinner hasn't shown up yet?
https://github.com/adobe/react-spectrum/pull/4900#discussion_r1303657303